lib/utils/sorted_list.dart

```diff
@@ -0,0 +1,93 @@
+abstract class AuList<T> {
+  late List<T> _list;
+
+  List<T> get items => _list;
```

Comment:
please use `List.unmodifiable(Iterable elements);`
We won't allow user to update the _list externally since it will mess up the order.